src/NzbDrone.Core/MediaFiles/EpisodeImport/Aggregation/Aggregators/AggregateLanguage.cs

```diff
@@ -1,71 +1,54 @@
         private readonly Logger _logger;
 
-        public AggregateLanguage(Logger logger)
+        public AggregateLanguage(IEnumerable<IAugmentLanguage> augmentLanguages, Logger logger)
         {
+            _augmentLanguages = augmentLanguages.OrderBy(a => a.Order).ToList();
             _logger = logger;
         }
 
         public LocalEpisode Aggregate(LocalEpisode localEpisode, DownloadClientItem downloadClientItem)
         {
-            // Get languages in preferred order, download client item, folder and finally file.
-            // Non-English languages will be preferred later, in the event there is a conflict
-            // between parsed languages the more preferred item will be used.
+            var languages = new List<Language> { };
+            var languagesConfidence = Confidence.Default;
 
-            var languages = new List<Language>
-                            {
-                                GetLanguage(localEpisode.DownloadClientEpisodeInfo, localEpisode.Episodes),
-                                GetLanguage(localEpisode.FolderEpisodeInfo, localEpisode.Episodes),
-                                GetLanguage(localEpisode.FileEpisodeInfo, localEpisode.Episodes)
-                            };
-
-            var language = languages.FirstOrDefault(l => l != Language.English) ?? Language.English;
-
-            _logger.Debug("Using language: {0}", language);
-
-            localEpisode.Language = language;
+            foreach (var augmentLanguage in _augmentLanguages)
+            {
+                var augmentedLanguage = augmentLanguage.AugmentLanguage(localEpisode, downloadClientItem);
+                if (augmentedLanguage == null)
+                {
+                    continue;
+                }
 
-            return localEpisode;
-        }
+                _logger.Trace("Considering Languages {0} ({1}) from {2}", string.Join(", ", augmentedLanguage.Languages ?? new List<Language>()), augmentedLanguage.Confidence, augmentLanguage.Name);
 
-        private Language GetLanguage(ParsedEpisodeInfo parsedEpisodeInfo, List<Episode> episodes)
-        {
-            if (parsedEpisodeInfo == null)
-            {
-                return Language.English;
+                if (augmentedLanguage?.Languages != null && augmentedLanguage.Languages.Count > 0 && !(augmentedLanguage.Languages.Count == 1 && augmentedLanguage.Languages.Contains(Language.Unknown)))
+                {
+                    languages = augmentedLanguage.Languages;
+                    languagesConfidence = augmentedLanguage.Confidence;
+                }
             }
 
-            var normalizedReleaseTitle = Parser.Parser.NormalizeEpisodeTitle(parsedEpisodeInfo.ReleaseTitle);
 
-            foreach (var episode in episodes)
-            {
-                var episodeTitleLanguage = LanguageParser.ParseLanguage(episode.Title, false);
+            var language = languages.FirstOrDefault(l => l != Language.English) ?? Language.English;
```

Comment:
This is going to cause issues some of the time. If you have a release that has 3 language tracks in the following order:

1. English
2. French
3. German

you're going to have issues because the language returned here is always going to be French, if you want German or English, it doesn't matter because this only sees French.

If the first non-Enlish track is `Unknown` then it'll come back as unknown.

Ideally this logic needs to be expanded so multiple audio tracks can be stored or in the mean time the first audio track should be used instead of some arbitrary one.

Reply:
Thank you for the review! For this particular one, although I can see your point, I didn't make any modification to that line. The behaviour you described already is what's happening on the released app.

As I mentioned in a comment above:

>  If you check the AggragateLanguage.cs at the line 48 (line 35 in the original file), returning the first non-english language we found was already what was done. I just "upgraded" the sources from where we fetch possible languages. I didn't change the decision about the selected language at the end.

So with this information, what do you want me to do? I know that ideally the real multi-language capability would have been the best but I honestly don't have that kind of skill. Does that mean this PR will never be accepted for it is? Just to know whether I should try to fix all the other point or should I cancel this PR right now.